samples/net/coap_client/src/main.c

```diff
@@ -0,0 +1,395 @@
+	int err, received;
+
+	wait_for_network();
+
+	if (server_resolve() != 0) {
+		LOG_ERR("Failed to resolve server name");
+		return;
+	}
+
+	if (client_init() != 0) {
+		LOG_ERR("Failed to initialize CoAP client");
+		return;
+	}
+
+	next_msg_time = k_uptime_get();
+
+	while (1) {
+		wait_for_network();
+
+		if (k_uptime_get() >= next_msg_time) {
+			if (client_get_send() != 0) {
+				LOG_ERR("Failed to send GET request, exit...");
+				break;
+			}
+
+			next_msg_time = k_uptime_get() + APP_COAP_SEND_INTERVAL_MS;
+		}
+
+		int64_t remaining = next_msg_time - k_uptime_get();
+
+		if (remaining < 0) {
+			remaining = 0;
+		}
+
+		err = wait(remaining);
+		if (err < 0) {
+			if (err == -EAGAIN) {
+				continue;
+			}
+
+			LOG_ERR("Poll error...");
+			continue;
+		}
+
+		received = recv(sock, coap_buf, sizeof(coap_buf), MSG_DONTWAIT);
+		if (received < 0) {
+			if (errno == EAGAIN || errno == EWOULDBLOCK) {
+				LOG_ERR("socket EAGAIN");
+				continue;
+			} else {
+				LOG_ERR("Socket error...");
+				continue;
+			}
+		}
+
+		if (received == 0) {
+			LOG_ERR("Empty datagram");
+			continue;
+		}
+
+		err = client_handle_get_response(coap_buf, received);
+		if (err < 0) {
+			LOG_ERR("Invalid response...");
+			continue;
+		}
+	}
+
+	(void)close(sock);
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb,
+			     uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_L4_CONNECTED:
+		LOG_INF("Network connectivity established");
+		k_mutex_lock(&network_connected_lock, K_FOREVER);
+		is_connected = true;
+		k_condvar_signal(&network_connected);
+		k_mutex_unlock(&network_connected_lock);
+		break;
+	case NET_EVENT_L4_DISCONNECTED:
+		LOG_INF("Network connectivity lost");
+		k_mutex_lock(&network_connected_lock, K_FOREVER);
+		is_connected = false;
+		k_mutex_unlock(&network_connected_lock);
+		break;
+	default:
+		/* Don't care */
+		return;
+	}
+}
+
+static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
+				       uint32_t event,
+				       struct net_if *iface)
+{
+	if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
+		LOG_ERR("NET_EVENT_CONN_IF_FATAL_ERROR");
+		k_fatal_halt(K_ERR_KERNEL_OOPS);
+		return;
+	}
+}
+
+int main(void)
+{
+	int err;
+
+	LOG_INF("The nRF CoAP client sample started");
+
+	/* Setup handler for Zephyr NET Connection Manager events. */
+	net_mgmt_init_event_callback(&l4_cb, l4_event_handler, L4_EVENT_MASK);
+	net_mgmt_add_event_callback(&l4_cb);
+
+	/* Setup handler for Zephyr NET Connection Manager Connectivity layer. */
+	net_mgmt_init_event_callback(&conn_cb, connectivity_event_handler, CONN_LAYER_EVENT_MASK);
+	net_mgmt_add_event_callback(&conn_cb);
+
+	/* Bring all network interfaces up.
+	 * Wi-Fi or LTE depending on the board that the sample was built for.
+	 */
+	LOG_INF("Bringing network interface up and connecting to the network");
+	err = conn_mgr_all_if_up(true);
+	if (err) {
+		LOG_ERR("conn_mgr_all_if_up, error: %d", err);
+		return err;
+	}
```

Comment:
Could we please add an explicit call to `conn_mgr_connect` ?

Reply:
Thanks, should be fixed now!